x-pack/plugins/security_solution/public/resolver/view/panels/panel_content_related_list.tsx

```diff
@@ -0,0 +1,226 @@
+import * as selectors from '../../store/selectors';
+import { useResolverDispatch } from '../use_resolver_dispatch';
+
+/**
+ * This view presents a list of related events of a given type for a given process.
+ * It will appear like:
+ *
+ * |                                                        |
+ * | :----------------------------------------------------- |
+ * | **registry deletion** @ *3:32PM..* *HKLM/software...*  |
+ * | **file creation** @ *3:34PM..* *C:/directory/file.exe* |
+ */
+
+interface MatchingEventEntry {
+  formattedDate: string;
+  eventType: string;
+  name: string;
+  entityId: string;
+  setQueryParams: () => void;
+}
+
+const DisplayList = memo(function DisplayList({
+  crumbs,
+  matchingEventEntries,
+}: {
+  crumbs: Array<{ text: string | JSX.Element; onClick: () => void }>;
+  matchingEventEntries: MatchingEventEntry[];
+}) {
+  return (
+    <>
+      <StyledBreadcrumbs breadcrumbs={crumbs} />
+      <EuiSpacer size="l" />
+      <>
+        {matchingEventEntries.map((eventView, index) => {
+          return (
+            <Fragment key={index}>
+              <EuiText>
+                <BoldCode>{eventView.eventType}</BoldCode>
+                {' @ '}
+                {eventView.formattedDate}
+              </EuiText>
+              <EuiSpacer size="xs" />
+              <EuiButtonEmpty onClick={eventView.setQueryParams}>{eventView.name}</EuiButtonEmpty>
+              {index === matchingEventEntries.length - 1 ? null : <EuiHorizontalRule margin="m" />}
+            </Fragment>
+          );
+        })}
+      </>
+    </>
+  );
+});
+
+export const ProcessEventListNarrowedByType = memo(function ProcessEventListNarrowedByType({
+  processEvent,
+  eventType,
+  relatedStats,
+  pushToQueryParams,
+}: {
+  processEvent: ResolverEvent;
+  pushToQueryParams: (arg0: CrumbInfo) => unknown;
+  eventType: string;
+  relatedStats: ResolverNodeStats;
+}) {
+  const processName = processEvent && event.eventName(processEvent);
+  const processEntityId = event.entityId(processEvent);
+  const totalCount = Object.values(relatedStats.events.byCategory).reduce(
+    (sum, val) => sum + val,
+    0
+  );
+  const eventsString = i18n.translate(
+    'xpack.securitySolution.enpoint.resolver.panel.processEventListByType.events',
+    {
+      defaultMessage: 'Events',
+    }
+  );
+  const waitingString = i18n.translate(
+    'xpack.securitySolution.enpoint.resolver.panel.processEventListByType.wait',
+    {
+      defaultMessage: 'Waiting For Events...',
+    }
+  );
+
+  const relatedsReadyMap = useSelector(selectors.relatedEventsReady);
+  const relatedsReady = relatedsReadyMap.get(processEntityId);
+
+  const relatedEventsForThisProcess = useSelector(selectors.relatedEventsByEntityId).get(
+    processEntityId
+  );
+  const dispatch = useResolverDispatch();
+
+  useEffect(() => {
+    if (typeof relatedsReady === 'undefined') {
+      dispatch({

```

Comment:
Let's discuss firing this from the middleware

Reply:
Yeah, I'd be interested in hearing what your thoughts are about that. I was a little reluctant to add things to reducer/selectors/middleware that derived from `window.location.history` based on statements you've made in the past to the effect that you'd prefer we _don't_ include UI-based dependencies there for the reason that it makes them harder to test/model/etc. I know the Book of Oatkiller says also to do things in selectors/reducers where possible, and I considered that - but I thought the direction to not put interface-dependent things in those places would override that in this case.

Reply:
let's sync in person.

Reply:
re: our discussion. Could you change this to a new action that starts w/ 'app'? ty

Reply:
https://github.com/elastic/kibana/pull/69098/commits/1058e680726ef3f49b925e610a67cf750a43c827